File: app/android/app/src/main/kotlin/com/mednote/MedNoteApp.kt
package com.mednote

import android.app.Application
import com.mednote.data.SettingsStore
import com.mednote.data.VisitStore
import com.mednote.service.AudioRecorderService
import com.mednote.service.VisitPipeline

class MedNoteApp : Application() {
    lateinit var settingsStore: SettingsStore
        private set
    lateinit var visitStore: VisitStore
        private set
    lateinit var audioRecorder: AudioRecorderService
        private set
    val visitPipeline = VisitPipeline()

    override fun onCreate() {
        super.onCreate()
        settingsStore = SettingsStore(this)
        visitStore = VisitStore(this)
        audioRecorder = AudioRecorderService(this)
    }
}

File: app/ios/MedNote/MedNoteApp.swift
import SwiftUI

@main
struct MedNoteApp: App {
    @StateObject private var settings = SettingsStore()
    @StateObject private var visitStore = VisitStore()
    @StateObject private var pipeline = VisitPipeline()
    @StateObject private var recorder = AudioRecorderService()

    var body: some Scene {
        WindowGroup {
            ContentView()
                .environmentObject(settings)
                .environmentObject(visitStore)
                .environmentObject(pipeline)
                .environmentObject(recorder)
        }
    }
}
